src/MagnumPlugins/ObjImporter/ObjImporter.cpp

```diff
@@ -115,387 +372,575 @@ ObjImporter::~ObjImporter() = default;
+                        positionIndices.push_back(vertex[0]);
+                        textureCoordinateIndices.push_back(vertex[1]);
+                        normalIndices.push_back(vertex[2]);
+
+                        hasTexCoords = hasTexCoords || (vertex[1] != 0);
+                        hasNormals = hasNormals || (vertex[2] != 0);
+
+                        pos = skipWhitespaces(pos, false);
+                    }
+                    pos = skipWhitespaces(ignoreLine(pos));
+                }
+                break;
+            case MeshPrimitive::Lines:
+                while(!pos.empty()) {
+                    CORRADE_ASSERT(pos[0] == 'l' && pos[1] == ' ', "Unexpected primitive keyword for Lines", {});
+                    pos = pos.suffix(2);
+
+                    while(!atEndOfLine(pos)) {
+                        std::array<int, 2> line;
+                        std::tie(line, pos) = parseLine(pos);
+                        positionIndices.push_back(line[0]);
+                        textureCoordinateIndices.push_back(line[1]);
+
+                        hasTexCoords = hasTexCoords || (line[1] != 0);
+
+                        pos = skipWhitespaces(pos, false);
+                    }
+                    pos = skipWhitespaces(ignoreLine(pos));
+                }
+                break;
+            case MeshPrimitive::Points:
+                while(!pos.empty()) {
+                    CORRADE_ASSERT(pos[0] == 'p' && pos[1] == ' ', "Unexpected primitive keyword for Points", {});
+                    pos = pos.suffix(2);
+
+                    ArrayView<char> word;
+                    while(!atEndOfLine(pos)) {
+                        std::tie(word, pos) = nextWord(pos);
+                        positionIndices.push_back(strToInt(word));
+                        pos = skipWhitespaces(pos, false);
+                    }
+                    pos = skipWhitespaces(ignoreLine(pos));
+                }
+                break;
+            default:
+                CORRADE_ASSERT_UNREACHABLE();
+        }
+    }
+
+    /* Merge index arrays, if there aren't just the positions */
+    std::vector<UnsignedInt> indices;
+
+    std::vector<std::vector<Vector3>> positionLayers;
+    std::vector<std::vector<Vector3>> normalLayers;
+    std::vector<std::vector<Vector2>> texCoordLayers;
+
+    std::vector<std::reference_wrapper<std::vector<UnsignedInt>>> arrays;
+    arrays.reserve(3);
+    arrays.emplace_back(positionIndices);
+    if(hasNormals) arrays.emplace_back(normalIndices);
+    if(hasTexCoords) arrays.emplace_back(textureCoordinateIndices);
+    indices = MeshTools::combineIndexArrays(arrays);
+
+    /* Reindex data arrays */
+    try {
+        positionLayers.push_back(reindex(positionIndices, _state->positions));
+        if(hasNormals) normalLayers.push_back(reindex(normalIndices, _state->normals));
+        if(hasTexCoords) texCoordLayers.push_back(reindex(textureCoordinateIndices, _state->texCoords));
```

Comment:
Sorry, the code was bad to begin with :/

I would just remove the `throw 0` from `reindex()` and return zero-sized vector in that case, then you can get rid of the `try` block completely.